presto-main/src/main/java/com/facebook/presto/memory/MemoryPool.java

```diff
@@ -351,6 +353,16 @@ private synchronized void updateTaggedMemoryAllocations(QueryId queryId, String
     @VisibleForTesting
     synchronized Map<QueryId, Map<String, Long>> getTaggedMemoryAllocations()
```

Comment:
This is getting quite expensive. Let's just change this to `getTaggedMemoryAllocations(queryId)` instead. If tests are not easy to fix, introduce a separate function instead so this one (with the old implementation) is only used in tests.